pensieve/config.py

```diff
@@ -26,31 +26,90 @@
 import jinja2
 import mozanalysis.metrics
 import mozanalysis.metrics.desktop
+import pandas
 
-from pensieve.analysis import AnalysisPeriod
+from . import AnalysisPeriod
+from pensieve.statistics import Statistic, BootstrapMean, StatisticResultCollection
 import pensieve.experimenter
 
 
-# TODO: these should be objects that pair a metric with a default statistical treatment.
+@attr.s(auto_attribs=True)
+class MetricWithTreatment:
+    """Represents a metric with a statistical treatment."""
+
+    metric: mozanalysis.metrics.Metric
+    treatment: Statistic
+
+    def run(self, data: pandas.DataFrame) -> "StatisticResultCollection":
+        """Apply the statistic transformation for data related to the specified metric."""
+        return self.treatment.apply(data, self.metric.name)
+
+
 DEFAULT_METRICS = {
     "desktop": {
-        "daily": [mozanalysis.metrics.desktop.unenroll],
-        "weekly": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.DAY: [
+            MetricWithTreatment(
+                metric=mozanalysis.metrics.desktop.unenroll,
+                treatment=BootstrapMean.from_config(
+                    {"num_samples": 1000, "branches": ["branch1", "branch2"]}
+                ),
+            )
         ],
-        "overall": [
-            mozanalysis.metrics.desktop.active_hours,
-            mozanalysis.metrics.desktop.uri_count,
-            mozanalysis.metrics.desktop.ad_clicks,
-            mozanalysis.metrics.desktop.search_count,
+        AnalysisPeriod.WEEK: [
+            MetricWithTreatment(
+                metric=mozanalysis.metrics.desktop.active_hours,
+                treatment=BootstrapMean.from_config(
```

Comment:
Right now there is only one statistic defined. I feel like adding more statistics should be a follow-up PR, then this list of standard treatments can also be updated.

Reply:
sgtm